src/autotype/AutoTypeSelectDialog.cpp

```diff
@@ -93,13 +86,23 @@ AutoTypeSelectDialog::~AutoTypeSelectDialog()
 {
 }
 
-void AutoTypeSelectDialog::setMatches(const QList<AutoTypeMatch>& matches, const QList<QSharedPointer<Database>>& dbs)
+void AutoTypeSelectDialog::setMatches(const QList<AutoTypeMatch>& matches,
+                                      const QList<QSharedPointer<Database>>& dbs,
+                                      const AutoTypeMatch& lastMatch)
 {
     m_matches = matches;
     m_dbs = dbs;
+    m_lastMatch = lastMatch;
+
+    // disable changing search scope if we have no direct matches
+    m_ui->searchCheckBox->setDisabled(m_matches.isEmpty());
 
-    m_ui->view->setMatchList(m_matches, !m_matches.isEmpty() || !m_ui->search->text().isEmpty());
-    m_ui->searchCheckBox->setChecked(m_matches.isEmpty());
+    // changing check also performs search
+    if (m_ui->searchCheckBox->isChecked() != m_matches.isEmpty()) {
+        m_ui->searchCheckBox->setChecked(m_matches.isEmpty());
```

Comment:
We use m_matches.isEmpty() three times, lets give it a variable name for ease of reading: `bool noMatches = m_matches.isEmpty()`

I also have to read this if statement several times to understand what is going on. I think we should just always set the search checkbox here to the noMatches boolean and get rid of the if statement. We could suppress the checkbox signals to prevent a double search.